File: src/main/java/com/tms/csp/transforms/OrToAnd.java
package com.tms.csp.transforms;

import com.tms.csp.ast.Exp;
import com.tms.csp.ast.Or;

/**
 *
 */
public class OrToAnd extends BaseTransformer {

    @Override
    protected Exp local(Exp in) {
        assert in.isOr();

//        if(true) throw new IllegalStateException();
        Or or = in.asOr();
        return or.createEquivAnd();
    }

    @Override
    protected boolean executeLocal(Exp in) {
        return in.isOr();
    }


}

File: src/main/java/com/tms/csp/transforms/OrToAnd.kt
package com.tms.csp.transforms

import com.tms.csp.ast.Exp

/**
 *
 */
class OrToAnd : BaseTransformer() {

    override fun local(inExp: Exp): Exp? {
        assert(inExp.isOr)

        //        if(true) throw new IllegalStateException();
        val or = inExp.asOr
        return or.createEquivAnd()
    }

    override fun executeLocal(expIn: Exp): Boolean {
        return expIn.isOr
    }


}
